commit 1a30a971311720f41862f7934f9b0fd947ec623f (refs/changes/31/5031/2)
Author: John Levon <john.levon@joyent.com>
Date:   2018-11-06T16:43:27+00:00 (11 months ago)
    
    OS-7336 lx_acl_getxattr leaks the ACL
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/usr/src/boot/sys/boot/efi/libefi/delay.c b/usr/src/boot/sys/boot/efi/libefi/delay.c
index c805fa13a7..0bd4339d54 100644
--- a/usr/src/boot/sys/boot/efi/libefi/delay.c
+++ b/usr/src/boot/sys/boot/efi/libefi/delay.c
@@ -1,4 +1,4 @@
-/*-
+/*
  * Copyright (c) 2001 Doug Rabson
  * All rights reserved.
  *
@@ -25,7 +25,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
 
 #include <efi.h>
 #include <efilib.h>
@@ -35,15 +34,5 @@ void delay(int);
 void
 delay(int usecs)
 {
-	static EFI_EVENT ev = 0;
-	UINTN junk;
-
-	if (!ev) {
-		if (BS->CreateEvent(EVT_TIMER, TPL_APPLICATION, 0, 0, &ev)
-		    != EFI_SUCCESS)
-			return;
-	}
-
-	BS->SetTimer(ev, TimerRelative, usecs * 10);
-	BS->WaitForEvent(1, &ev, &junk);
+	BS->Stall(usecs);
 }
diff --git a/usr/src/uts/common/brand/lx/os/lx_acl.c b/usr/src/uts/common/brand/lx/os/lx_acl.c
index 594671f401..184f05b6ed 100644
--- a/usr/src/uts/common/brand/lx/os/lx_acl.c
+++ b/usr/src/uts/common/brand/lx/os/lx_acl.c
@@ -10,7 +10,7 @@
  */
 
 /*
- * Copyright (c) 2017 Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 #include <sys/types.h>
@@ -192,6 +192,9 @@ lx_acl_getxattr(vnode_t *vp, enum lx_acl_type atype, void *data, size_t slen,
 		return (err);
 	}
 
+	if (vsattr.vsa_aclentp != NULL)
+		kmem_free(vsattr.vsa_aclentp, vsattr.vsa_aclentsz);
+
 	return (ENODATA);
 }
 
